syntax = "proto3";

package user;

option go_package = "internal/proto;userpb";

service UserService {
  rpc GetUserById(GetUserByIdRequest) returns (GetUserByIdResponse);
  rpc CreateIfNotExists(CreateIfNotExistsRequest) returns (CreateIfNotExistsResponse);
  rpc GetUserInventory(GetUserInventoryRequest) returns (GetUserInventoryResponse);
}

message GetUserByIdRequest {
  string id = 1;
}

message GetUserByIdResponse {
  string id = 1;
  string username = 2;
  string hashed_password = 3;
}

message CreateIfNotExistsRequest {
  string username = 1;
  string password = 2;
}

message CreateIfNotExistsResponse {
  string id = 1;
  string username = 2;
  string hashed_password = 3;
  bool is_new = 4;  // True, если пользователь создан
}

message GetUserInventoryRequest {
  string username = 1;
}

message GetUserInventoryResponse {
  repeated InventoryItem items = 1;
}

message InventoryItem {
  string merch = 1;
  int32 quantity = 2;
}
